### PR TITLE
Fixes #9889: Remove common conf between relay and webapp from webapp package

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/9889

### CODE REVIEW

**Comment by amousset** [comment]
Still WIP for the packaging part.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Still missing:
* apache modules configuration
* vhost migration

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
@peckpeck Should be complete now.

File: rudder-webapp/SOURCES/rudder-webapp-apache [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Apache modules
```
Comment (amousset) [inline rudder-webapp/SOURCES/rudder-webapp-apache]: Moved from rudder repo to keep all apache module for rudder-webapp in the same repo.

File: rudder-server-relay/debian/rules [hunk 2]
```diff
@@ -55,10 +55,8 @@ binary-arch: install
 	dh_installchangelogs
 #	dh_installdocs
 #	dh_installexamples
-	cp $(CURDIR)/SOURCES/rudder-relay-vhost.conf $(CURDIR)/BUILD/rudder-relay-vhost
-	cp $(CURDIR)/SOURCES/rudder-relay-vhost-ssl.conf $(CURDIR)/BUILD/rudder-relay-vhost-ssl
-	dh_install --SOURCEDIR=$(CURDIR)/BUILD/ rudder-relay-vhost /etc/apache2/sites-available/
-	dh_install --SOURCEDIR=$(CURDIR)/BUILD/ rudder-relay-vhost-ssl /etc/apache2/sites-available/
+	cp $(CURDIR)/SOURCES/rudder-relay-vhost.conf $(CURDIR)/BUILD/rudder
+	dh_install --SOURCEDIR=$(CURDIR)/BUILD/ rudder /etc/apache2/sites-available/rudder
 	dh_install --SOURCEDIR=$(CURDIR)/SOURCES/ rudder-relay-apache-common.conf /opt/rudder/etc/
```
Comment (peckpeck) [inline rudder-server-relay/debian/rules]: To change

File: rudder-server-relay/SPECS/rudder-server-relay.spec [hunk 3]
```diff
@@ -142,8 +141,7 @@ cp %{_sourcedir}/relay-api/apache/relay-api.wsgi %{buildroot}%{rudderdir}/share/
 install -m 644 %{_sourcedir}/relay-api/apache/relay-api.conf %{buildroot}/etc/%{apache_vhost_dir}/relay-api.conf
 
 # Others
-install -m 644 %{SOURCE1} %{buildroot}/etc/%{apache_vhost_dir}/rudder-relay-vhost.conf
-install -m 644 %{SOURCE4} %{buildroot}/etc/%{apache_vhost_dir}/rudder-relay-vhost-ssl.conf
+install -m 644 %{SOURCE1} %{buildroot}/etc/%{apache_vhost_dir}/rudder.conf
 install -m 644 %{SOURCE5} %{buildroot}%{rudderdir}/etc/rudder-relay-apache-common.conf
 install -m 644 %{SOURCE6} %{buildroot}/etc/sysconfig/rudder-relay-apache
```
Comment (peckpeck) [inline rudder-server-relay/SPECS/rudder-server-relay.spec]: to change

File: rudder-server-relay/SPECS/rudder-server-relay.spec [hunk 4]
```diff
@@ -77,7 +76,7 @@ BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
 ## General
 BuildRequires: python, python-devel
-Requires: rudder-agent, rsyslog, openssl, %{apache}, %{apache_tools}, python
+Requires: rudder-agent = %{real_epoch}:%{real_version}, rsyslog, openssl, %{apache}, %{apache_tools}, python
```
Comment (peckpeck) [inline rudder-server-relay/SPECS/rudder-server-relay.spec]: I don't think we should have an agent version constraint

File: rudder-server-relay/SPECS/rudder-server-relay.spec [hunk 5]
```diff
@@ -190,17 +188,22 @@ done
 
 # Do this ONLY at first install
 if [ $1 -eq 1 ];  then
-  echo -e '# This sources the configuration file needed by Rudder\n. /etc/sysconfig/rudder-relay-apache' >> /etc/sysconfig/apache2
   echo 'DAVLockDB /tmp/davlock.db' > /etc/%{apache}/conf.d/dav_mod.conf
 fi
 
-# Add required includes in the SLES apache2 configuration
-%if 0%{?suse_version}
```
Comment (peckpeck) [inline rudder-server-relay/SPECS/rudder-server-relay.spec]: You should keep the %suse

File: rudder-server-relay/SPECS/rudder-server-relay.spec [hunk 6]
```diff
@@ -234,7 +259,7 @@ service %{apache} start > /dev/null && echo " Done"
 /bin/systemctl start  %{apache}.service && echo " Done"
 %endif
 
-
+# TODO : do not display this on webapp
```
Comment (peckpeck) [inline rudder-server-relay/SPECS/rudder-server-relay.spec]: Add this information in the comment: if this is not the root server you have to ...

File: rudder-server-relay/debian/control [hunk 7]
```diff
@@ -8,8 +8,8 @@ Homepage: http://www.rudder-project.org
 
 Package: rudder-server-relay
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, python, rudder-agent, apache2, apache2-utils, rsyslog, openssl, libapache2-mod-wsgi
+Depends: ${shlibs:Depends}, ${misc:Depends}, python, rudder-agent (>= ${binary:Version}), apache2, apache2-utils, rsyslog, openssl, libapache2-mod-wsgi
```
Comment (peckpeck) [inline rudder-server-relay/debian/control]: I don't think we should depend on a specific agent version except for a good reason

File: rudder-server-relay/debian/postinst [hunk 8]
```diff
@@ -67,18 +67,41 @@ case "$1" in
...
+          mv /etc/apache2/sites-available/${OLD_VHOST} /var/backups/${OLD_VHOST}-$(date +%s)
+          echo " Done"
+        fi
+      done
+
       echo -n "INFO: Restarting Apache HTTPd..."
       service apache2 restart >/dev/null 2>&1
       echo " Done"
 
+      # TODO
```
Comment (peckpeck) [inline rudder-server-relay/debian/postinst]: idem

File: rudder-webapp/SOURCES/rudder-networks-24.conf [hunk 9]
```diff
@@ -1 +0,0 @@
-Require all denied
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-networks-24.conf]: Th matching file is missing from the relay package

File: rudder-webapp/SPECS/rudder-webapp.spec [hunk 10]
```diff
@@ -109,12 +99,13 @@ Source20: rudder-webapp.te
...
 Source22: rudder-keys
 Source23: .gitignore
+Source24: rudder-webapp-apache
 
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 BuildArch: noarch
 
 # Dependencies
-Requires: rudder-techniques = %{real_epoch}:%{real_version}, ncf, ncf-api-virtualenv, %{apache}, %{apache_tools}, git-core, rsync, openssl, %{ldap_clients}
+Requires: rudder-techniques = %{real_epoch}:%{real_version}, rudder-server-relay = %{real_epoch}:%{real_version}, ncf, ncf-api-virtualenv, %{apache}, %{apache_tools}, git-core, rsync, openssl, %{ldap_clients}
```
Comment (peckpeck) [inline rudder-webapp/SPECS/rudder-webapp.spec]: idem

File: rudder-webapp/SPECS/rudder-webapp.spec [hunk 11]
```diff
@@ -245,19 +233,16 @@ cp %{_builddir}/rudder-sources/rudder/rudder-web/target/rudder-web*.war %{buildr
...
 install -m 644 %{SOURCE2} %{buildroot}%{rudderdir}/share/webapps/
 
-# Copy stub rudder-networks*.conf
-cp %{SOURCE3} %{buildroot}%{rudderdir}/etc/
-cp %{SOURCE4} %{buildroot}%{rudderdir}/etc/
-
+# TODO ??
 %if 0%{?suse_version}
 # On SLES, change the Apache DocumentRoot to the OS default
 sed -i "s%^DocumentRoot /var/www$%DocumentRoot /srv/www%" %{buildroot}%{rudderdir}/etc/rudder-apache-common.conf
```
Comment (peckpeck) [inline rudder-webapp/SPECS/rudder-webapp.spec]: rename the file

**Comment by amousset** [comment]
Commit modified

**Pull request merged**